### ISSUE TITLE
Android react-native sometimes optimizes and removes View as single child element of custom ScrollView, which causes an exception

### ISSUE THREAD

**Issue by jblack10101** [facebook/react-native#5799]
In Android, ScrollViews are only allowed to have a single child element.  I created a custom component that extends ScrollView, and I wanted to setup my layout like this:

```
<CustomScrollView>
    <View>
        <Text>text</Text>
        <TextInput />
    </View>
</CustomScrollView>
```

I found that if I didn't add any styling or anything to that View, that it would get optimized out and removed from the view hierarchy, which would cause an exception ("ScrollView can host only one direct child").  I now know that adding a backgroundColor style will prevent it from being optimized out, but I haven't investigated what will prevent the optimization.

So basically, the optimization code needs to check whether the parent is a ScrollView, and if it is, don't optimize out any views.

I'm using the latest version in master as of a couple days prior to this posting (https://github.com/facebook/react-native/commit/736e90053243a03c48c00f072a2f3bbbdcb07567), which is after released version 0.19.

**Comment by howanghk** [comment]
try setting `collapsable` to false.

```
<CustomScrollView>
    <View collapsable={false}>
        <Text>text</Text>
        <TextInput />
    </View>
</CustomScrollView>
```

Reference from https://facebook.github.io/react-native/docs/view.html#collapsable

> Views that are only used to layout their children or otherwise don't draw anything may be automatically removed from the native hierarchy as an optimization.

**Comment by jblack10101** [comment]
This worked, thanks!
